src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -287,7 +287,39 @@ export class TerminalService implements ITerminalService {
 
 	public getAvailableProfiles(): ITerminalProfile[] {
 		this._updateAvailableProfiles();
-		return this._availableProfiles?.map(s => ({ profileName: s.profileName, path: s.path, args: s.args } as ITerminalProfile)) || [];
+		return this._availableProfiles?.map(s => ({ profileName: s.profileName, path: s.path, args: s.args, isWorkspaceProfile: this._getWorkspaceProfilePermissions(s) } as ITerminalProfile)) || [];
+	}
+
+	private _getWorkspaceProfilePermissions(profile: ITerminalProfile): boolean {
+		if (isWindows) {
+			const profiles = this._configurationService.inspect<{ [key: string]: ITerminalProfileObject }>(`terminal.integrated.profiles.windows`);
+			if (profiles && profiles.workspaceValue && profiles.defaultValue) {
+				const workspaceProfile = Object.entries(profiles.workspaceValue).find(p => p[0] === profile.profileName);
+				const defaultProfile = Object.entries(profiles.defaultValue).find(p => p[0] === profile.profileName);
+				if (workspaceProfile && defaultProfile && workspaceProfile[0] === defaultProfile[0]) {
+					let result = !this._terminalProfileObjectEqual(workspaceProfile[1], defaultProfile[1]);
+					return result;
+				} else {
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+	private _terminalProfileObjectEqual(one?: ITerminalProfileObject, two?: ITerminalProfileObject): boolean {
+		if (one === null && two === null) {
+			return true;
+		} else if (one && two && (one as ITerminalExecutable).path && (two as ITerminalExecutable).path) {
```

Comment:
No need to check one && too since we return above

```suggestion
		if (one === null && two === null) {
			return true;
		}
		if ('path' in one && 'path' in two) {
```